src/index.rs

```diff
@@ -47,6 +53,21 @@ impl Index {
 
         let field_id =
             unsafe { raw::RediSearch_CreateField(self.inner, name.as_ptr(), ftype.bits, fopt) };
+        unsafe {
+            raw::RediSearch_TextFieldSetWeight(self.inner, field_id, weight);
+            if let Some(options) = tag_options {
+                if let Some(sperator) = options.tag_separator {
+                    raw::RediSearch_TagFieldSetSeparator(self.inner, field_id, sperator as i8);
+                }
+                if options.tag_case_sensitive {
+                    raw::RediSearch_TagFieldSetCaseSensitive(
+                        self.inner,
+                        field_id,
+                        options.tag_case_sensitive as i32,
```

Comment:
Same here: Use `c_int`, not `i32`. These are FFI types.

Reply:
:+1: